src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheck.java

```diff
@@ -106,15 +134,56 @@ public void setIgnoreOverridden(boolean ignoreOverridden) {
     @Override
     protected boolean mustCheckName(DetailAST ast) {
         boolean checkName = true;
+        final boolean isDefault = scope == Scope.ANONINNER && excludeScope == null;
+
         if (ignoreOverridden && isOverriddenMethod(ast)
                 || ast.getParent().getType() == TokenTypes.LITERAL_CATCH
-                || CheckUtils.isReceiverParameter(ast)) {
+                || CheckUtils.isReceiverParameter(ast)
+                || !isDefault && !matchScope(calculateScope(ast))) {
             checkName = false;
         }
         return checkName;
     }
 
     /**
+     * Returns the scope for the method/constructor at the specified AST. If
+     * the method is in an interface or annotation block, the scope is assumed
+     * to be public.
+     *
+     * @param ast the token of the method/constructor
+     * @return the scope of the method/constructor
+     */
+    private static Scope calculateScope(final DetailAST ast) {
+        final DetailAST params = ast.getParent();
+        final DetailAST meth = params.getParent();
+        Scope scope = Scope.PRIVATE;
```

Comment:
@romani This is the scope of e.g lambda function parameters. Strictly speaking, the default should have been  'Scope.ANONINNER', however in light of your comments, I chose to use 'Scope.PRIVATE'. Please tell me if that's ok for you.